hack/make/.ensure-syscall-test

```diff
@@ -4,5 +4,20 @@ set -e
 # Build a C binary for cloning a userns for seccomp tests
 # and compile it for target daemon
 if [ "$DOCKER_ENGINE_GOOS" = "linux" ]; then
-	docker build ${DOCKER_BUILD_ARGS} -qt syscall-test contrib/syscall-test > /dev/null
+	if [ "$DOCKER_ENGINE_GOARCH" = "$DOCKER_CLIENT_OSARCH" ]; then
```

Comment:
@tianon I've made that change based on your comment https://github.com/docker/docker/pull/19504#discussion_r50461710. Could you tell me exactly how that should look so I can update it, please?